core/controllers/suggestion.py

```
@@ -108,8 +108,7 @@ def post(self):
             self.payload.get('target_type'), self.payload.get('target_id'),
```

Can you please add a check for theses values above?

---

The approach I took was to marshal the values into the domain object and call `validate` before the datastore PUT.

---

But that if validation fails we will throw a 500 error but we should throw a 400 error for invalid input right?

---

I see. I updated this to catch ValidationErrors and throw InvalidInput 400 errors eternally to the user. Does that work?